stdlib/universe/window.go

```diff
@@ -465,12 +459,8 @@ func hasValidPredecessors(node plan.PlanNode) bool {
 		DuplicateKind,
 		RenameKind,
 		ShiftKind,
-		SkewKind,
 		SortKind,
-		SpreadKind,
 		StateTrackingKind,
-		StddevKind,
-		SumKind,
 		UniqueKind:
```

Comment:
I got confused by the logic here because `go` does not have implicit fallthrough.

I think if you hit a `range` here, you can just return true and stop recursing.  Range will set the `_start` and `_stop` columns to be in the group key and have the same value in all tables.

Also, not all sources will produce uniform bound for all tables...  It would be possible to produce overlapping bounds using `csv.from` for example.

I think the best way to do this would be to check that the plan node implements `BoundsAwareProcedureSpec`, and that its `TimeBounds` method returns a non-nil value.  This will match both `range` and `influxdb.from` with bounds pushed down.

Reply:
I added this comment here because I was unsure if that was a correct assumption:
https://github.com/influxdata/flux/blob/3dd4a1fedbd8d5c578a7e6c21d89db988aaf9c66/stdlib/universe/window.go#L433

The time bounds check however will ensure it is. Thanks,